lib/Rudder.php

```diff
@@ -12,9 +12,45 @@ class Rudder {
    */
   public static function init($secret, $options = array()) {
     self::assert($secret, "Rudder::init() requires secret");
+    // check if ssl is here --> check if it is http or https
+    $this->handleSSL($options);
     self::$client = new Rudder_Client($secret, $options);
   }
 
+  /**
+   * checks the dataplane url format only is ssl key is present
+   * @param  array  $options  passed straight to the client
+   */
+  private function handleSSL($options) {
+    if(filter_var($options["data_plane_url"], FILTER_VALIDATE_URL)) {
+    if($options["ssl"] == 'true') { // if ssl is true only https is expected
+      if (isset($options["data_plane_url"])) {
+          $options["data_plane_url"] = $this->handleUrl($options["data_plane_url"],'https');  
+      }
+    } else {
+      // if ssl is false only http is expected
+      if (isset($options["data_plane_url"])) {
+        $options["data_plane_url"] = $this->handleUrl($options["data_plane_url"],'http');  
+      }
+    } }
```

Comment:
Let's refactor this logic a bit (also combined the above suggestion) like below,

```suggestion
    if(<validate data plane input here>) {
      $protocol = "https";
      if($options["ssl"] == false) {
        $protocol = "http";
      }
      
      $options["data_plane_url"] = $this->handleUrl($options["data_plane_url"], $protocol);
    }
```

Reply:
done